src/ui/controls/edit.cpp

```diff
@@ -90,11 +89,10 @@ bool IsSep(int character)
 //! Object's constructor.
 CEdit::CEdit()
     : CControl(),
+      m_text( CHARACTERS_MAX + 1, '\0' ),
```

Comment:
You are allocating 1MB of memory every time CEdit is created, this is effectively no better than an array with hardcoded size. Just leave it as an empty std::string object. Then you should be able to remove the CHARACTERS_MAX constant completly.